lib/internal/Magento/Framework/App/Test/Unit/MaintenanceModeTest.php

```diff
@@ -95,7 +98,7 @@ public function testisOnWithIP()
         $this->flagDir->expects($this->exactly(2))
             ->method('isExist')
             ->willReturnMap($mapisExist);
-        $this->assertFalse($this->model->isOn());
+        $this->assertTrue($this->model->isOn());
```

Comment:
It appears that this test is testing `!in_array("", [""])` which results in `false`.
So if your IP is `""` (the default value of the first and only `isON()` parameter) and the list of whitelisted IP's is `[""]` then you are marked as whitelisted and maintenance is not on for you. This is wrong.
If maintenance mode is on and the ip whitelist file is present but empty, maintenance mode should be on for everyone and not off. So this test is incorrect. I will fix this by adding an `array_filter` around the whitelisted IP's so that you can never compare against `[""]`, but in that case against `[]`, which will never have a false positive on `in_array` comparisons.